src/main/java/org/prebid/server/bidder/adoppler/AdopplerBidder.java

```diff
@@ -0,0 +1,189 @@
+            return Result.emptyWithError(BidderError.badInput("bad request"));
+        } else if (statusCode != HttpResponseStatus.OK.code()) {
+            return Result.emptyWithError(BidderError.badServerResponse(String.format("Unexpected HTTP status %s.",
+                    statusCode)));
+        }
+
+        final BidResponse bidResponse;
+        try {
+            bidResponse = decodeBodyToBidResponse(httpCall);
+        } catch (PreBidException e) {
+            return Result.emptyWithError(BidderError.badServerResponse(String.format("invalid body: %s",
+                    e.getMessage())));
+        }
+
+        final Map<String, BidType> impTypes = new HashMap<>();
+        for (Imp imp : bidRequest.getImp()) {
+            if (impTypes.get(imp.getId()) != null) {
+                return Result.emptyWithError(BidderError.badInput(String.format("duplicate $.imp.id %s", imp.getId())));
+            }
+            if (imp.getBanner() != null) {
+                impTypes.put(imp.getId(), BidType.banner);
+            } else if (imp.getVideo() != null) {
+                impTypes.put(imp.getId(), BidType.video);
+            } else if (imp.getAudio() != null) {
+                impTypes.put(imp.getId(), BidType.audio);
+            } else if (imp.getXNative() != null) {
+                impTypes.put(imp.getId(), BidType.xNative);
+            } else {
+                return Result.emptyWithError(BidderError.badInput(String.format(
+                        "one of $.imp.banner, $.imp.video, $.imp.audio and $.imp.native field required")));
+            }
+        }
+
+        final List<BidderBid> bidderBids = new ArrayList<>();
+        for (SeatBid seatBid : bidResponse.getSeatbid()) {
+            for (Bid bid : seatBid.getBid()) {
+                if (impTypes.get(bid.getImpid()) == null) {
+                    return Result.emptyWithError(BidderError
+                            .badServerResponse(String.format("unknown impid: %s", bid.getImpid())));
+                }
+                final AdopplerResponseExt adopplerResponseExt;
+                if (impTypes.get(bid.getImpid()) == BidType.video) {
+                    final ObjectNode ext = bid.getExt();
+                    try {
+                        adopplerResponseExt = parseResponseExt(ext);
+                    } catch (PreBidException e) {
+                        return Result.emptyWithError(BidderError.badServerResponse(e.getMessage()));
+                    }
+                    final AdopplerResponseVideoExt adopplerResponseVideoExt = adopplerResponseExt.getAds();
+                    if ((adopplerResponseExt == null) || (adopplerResponseVideoExt == null)) {
+                        return Result.emptyWithError(BidderError.badServerResponse(String
+                                .format("$.seatbid.bid.ext.ads.video required")));
+                    }
+                    ExtBidPrebidVideo.of(adopplerResponseVideoExt.getVideo().getDuration(), head(bid.getCat()));
```

Comment:
As you can see, we do not have separate place for BidVideo. (thats why this code is unnecessary) But you can exclude it into separate methods to use it in a future